test/unit-tests/running/test-runner/run-test.spec.ts

```diff
@@ -30,6 +31,79 @@ export class RunTestTests {
       Expect(outputStream.push).toHaveBeenCalledWith("ok 1 Test Function\n");
    }
 
+    @AsyncTest()
+    public singlePassingTestRunsSuccessfullyWithOnCompleteEventRaised() {
+        let testCompletedValue: ITestCompleteEvent = null;
+        const test = new TestBuilder().withTestCaseCount(1).build();
+
+        const testFixture = new TestFixtureBuilder()
+            .addTest(test)
+            .build();
+
+        const testSet = new TestSetBuilder()
+            .addTestFixture(testFixture)
+            .build();
+
+        const outputStream = new TestOutputStream();
+
+        const testRunner = new TestRunner(outputStream);
+
+        let spyContainer = {
+            onCompleteCB: (testCompleted: ITestCompleteEvent) => {
+                testCompletedValue = testCompleted;
+            }
+        };
+
+        testRunner.onTestComplete(spyContainer.onCompleteCB);
+
```

Comment:
If we could refactor the last sections of these tests to:

```typescript
await testRunner.run(testSet);
Expect(testCompletedValue).not.toBeNull();
Expect(testCompletedValue.currentTestIndex).toEqual(1);
```

That'd be awesome as with the little upgrades we can do that now :)